src/diffusers/loaders.py

```diff
@@ -962,72 +1047,68 @@ def lora_scale(self) -> float:
-                for _, text_encoder_attr in self._lora_attn_processor_attr_to_text_encoder_attr.items():
-                    # Retrieve the q/k/v/out projection of CLIPAttention
-                    module = attn_module.get_submodule(text_encoder_attr)
-                    if hasattr(module, "old_forward"):
-                        # restore original `forward` to remove monkey-patch
-                        module.forward = module.old_forward
-                        delattr(module, "old_forward")
-
-    def _modify_text_encoder(self, attn_processors: Dict[str, LoRAAttnProcessor]):
+    @classmethod
+    def _remove_text_encoder_monkey_patch(cls, text_encoder):
+        for _, attn_module in text_encoder_attn_modules(text_encoder):
+            if isinstance(text_encoder, CLIPTextModel):
+                if isinstance(attn_module.q_proj, PatchedLoraProjection):
+                    attn_module.q_proj = attn_module.q_proj.regular_linear_layer
+                    attn_module.k_proj = attn_module.k_proj.regular_linear_layer
+                    attn_module.v_proj = attn_module.v_proj.regular_linear_layer
+                    attn_module.out_proj = attn_module.out_proj.regular_linear_layer
+            elif isinstance(text_encoder, T5EncoderModel):
+                if isinstance(attn_module.q, PatchedLoraProjection):
+                    attn_module.q = attn_module.q.regular_linear_layer
+                    attn_module.k = attn_module.k.regular_linear_layer
+                    attn_module.v = attn_module.v.regular_linear_layer
+                    attn_module.o = attn_module.o.regular_linear_layer
+            else:
+                raise ValueError(f"{text_encoder.__class__.__name__} does not support LoRA training")
+
+    @classmethod
+    def _modify_text_encoder(cls, text_encoder, lora_scale=1, network_alpha=None):
         r"""
         Monkey-patches the forward passes of attention modules of the text encoder.
-
-        Parameters:
-            attn_processors: Dict[str, `LoRAAttnProcessor`]:
-                A dictionary mapping the module names and their corresponding [`~LoRAAttnProcessor`].
         """
 
         # First, remove any monkey-patch that might have been applied before
-        self._remove_text_encoder_monkey_patch()
+        cls._remove_text_encoder_monkey_patch(text_encoder)
 
-        # Loop over the CLIPAttention module of text_encoder
-        for name, attn_module in self.text_encoder.named_modules():
-            if name.endswith(TEXT_ENCODER_ATTN_MODULE):
-                # Loop over the LoRA layers
-                for attn_proc_attr, text_encoder_attr in self._lora_attn_processor_attr_to_text_encoder_attr.items():
-                    # Retrieve the q/k/v/out projection of CLIPAttention and its corresponding LoRA layer.
-                    module = attn_module.get_submodule(text_encoder_attr)
-                    lora_layer = attn_processors[name].get_submodule(attn_proc_attr)
+        lora_parameters = []
```

Comment:
Return the new parameters which makes them easy to pass to the optimizer in the training script